### DISCUSSION TITLE
How to cycle through previous commit messages in the commit editor?

### DISCUSSION THREAD

**Discussion by azzamsa** [NeogitOrg/neogit#1079] [Q&A] [answered]
Hi, 👋.

In Magit, I can use `M-p`  and `M-n` to cycle through previous commit history.
This is very useful especially when the current commit message is similar to the previous commit message.

My workflow would be:

1. `M-p` for `(git-commit-prev-message)`
2. Make a small edit
3. Commit

Do we have a similar feature?

Thanks for Neogit! 🎖️

**Comment by CKolkey** [comment, marked as answer]
We do not 😅 (yet). Am I reading it (magit docs/source) right that its per-session? It doesn't look like it's pulling it from git, but rather an internal buffer.

**Reply by azzamsa** [reply]
>  that its per-session? 

Yes, correct. It is only per-session.

**Reply by CKolkey** [reply]
How does this behaviour compare? https://github.com/NeogitOrg/neogit/pull/1083 It's.. seemingly... pretty easy to just pull from git.
